src/binder/bind/bind_copy.cpp

```diff
@@ -16,25 +16,78 @@ std::unique_ptr<BoundStatement> Binder::bindCopy(const Statement& statement) {
     validateTableExist(catalog, tableName);
     auto tableID = catalogContent->getTableID(tableName);
     auto csvReaderConfig = bindParsingOptions(copyCSV.getParsingOptions());
-    auto filePaths = bindFilePaths(copyCSV.getFilePaths());
-    auto fileType = bindFileType(filePaths);
+    auto boundFilePaths = bindFilePaths(copyCSV.getFilePaths());
+    auto fileType = bindFileType(boundFilePaths);
+    std::vector<std::string> filePaths;
+    std::unordered_map<common::property_id_t, std::string> propertyToNpyMap;
+    if (fileType == CopyDescription::FileType::NPY) {
+        propertyToNpyMap =
+            bindPropertyToNpyMap(tableID, boundFilePaths, copyCSV.getPropertyNames());
+    } else {
+        filePaths = boundFilePaths[0];
+    }
     return make_unique<BoundCopy>(
-        CopyDescription(filePaths, csvReaderConfig, fileType), tableID, tableName);
+        CopyDescription(filePaths, propertyToNpyMap, csvReaderConfig, fileType), tableID,
+        tableName);
 }
 
-std::vector<std::string> Binder::bindFilePaths(const std::vector<std::string>& filePaths) {
-    std::vector<std::string> boundFilePaths;
-    for (auto& filePath : filePaths) {
-        auto globbedFilePaths = FileUtils::globFilePath(filePath);
-        boundFilePaths.insert(
-            boundFilePaths.end(), globbedFilePaths.begin(), globbedFilePaths.end());
-    }
-    if (boundFilePaths.empty()) {
-        throw BinderException{StringUtils::string_format("Invalid file path: {}.", filePaths[0])};
+std::vector<std::vector<std::string>> Binder::bindFilePaths(
+    const std::vector<std::vector<std::string>>& filePaths) {
+    std::vector<std::vector<std::string>> boundFilePaths;
+    for (auto& currPaths : filePaths) {
+        std::vector<std::string> currBoundFilePaths;
+        for (auto& filePath : currPaths) {
+            auto globbedFilePaths = FileUtils::globFilePath(filePath);
+            currBoundFilePaths.insert(
+                currBoundFilePaths.end(), globbedFilePaths.begin(), globbedFilePaths.end());
+            if (currBoundFilePaths.empty()) {
+                throw BinderException{
+                    StringUtils::string_format("Invalid file path: {}.", currPaths[0])};
+            }
+        }
+        boundFilePaths.emplace_back(currBoundFilePaths);
     }
     return boundFilePaths;
 }
 
+std::unordered_map<common::property_id_t, std::string> Binder::bindPropertyToNpyMap(
```

Comment:
this is up to u. But I think you can simply bind to vector<string> so that it aligns with the order in schema. Also we don't need to support multiple npy files to 1 column as an experimental feature.